app/main.py

```diff
@@ -1,3 +1,133 @@
+    butter = costumer.butter() * shop.butter_price()
+    distance = distance_to_shop(costumer, shop.location)
+    ride_cost = consumption_for_the_ride(
+        distance, costumer.car_fuel_consumption(), fuel_price
+    )
+    sum_purchase = milk + bread + butter + ride_cost
+    return round(sum_purchase, 2)
+
+
+def shop_trip() -> None:
+    list_of_costumers_instance = []
+    list_of_shops_instance = []
+    result_info_for_printing = []
+
+    with open("app/config.json", "r") as file:
+        result = json.load(file)
+
+    list_of_customers = result.get("customers")
+    fuel_price = result.get("FUEL_PRICE")
+    shops_list = result.get("shops")
+
+    for costumer in list_of_customers:
+        list_of_costumers_instance.append(Customer(
+            costumer["name"], costumer["product_cart"],
+            costumer["location"], costumer["money"], costumer["car"]
+        ))
+
+    for shop in shops_list:
+        list_of_shops_instance.append(Shop(
+            shop["name"], shop["location"], shop["products"]
+        ))
+
+    for costumer in list_of_costumers_instance:
+        total_shops = []
+        shop_name = {}
+        grocery = {}
+
+        result_info_for_printing.append(costumer.__str__())
+
+        for shop in list_of_shops_instance:
+            sum_of_consumption = consumption_for_purchase(
+                costumer, shop, fuel_price
+            )
+            total_shops.append(sum_of_consumption)
+            shop_name[sum_of_consumption] = shop.name
+            grocery[shop.name] = {
+                "milk": shop.milk_price(),
+                "bread": shop.bread_price(),
+                "butter": shop.butter_price(),
+                "location": shop.location
+            }
+            result_info_for_printing.append(
+                f"{costumer.name}'s trip to the {shop.name}"
+                f" costs {sum_of_consumption}"
+            )
+        cheapest_shop = min(total_shops)
+        current_shop = shop_name[cheapest_shop]
+        current_date_time = datetime.datetime.now()
+        time_now = current_date_time.strftime("%d/%m/%Y %H:%M:%S")
+
+        if cheapest_shop > costumer.money:
+            result_info_for_printing.append(
+                f"{costumer.name} doesn't have enough money "
+                f"to make a purchase in any shop"
+            )
+            continue
+        result_info_for_printing.append(
+            f"{costumer.name} rides to {current_shop}"
+        )
+        result_info_for_printing.append("")
+        result_info_for_printing.append(f"Date: {time_now}")
+        result_info_for_printing.append(
+            f"Thanks, {costumer.name}, for your purchase!"
+        )
+        total_milk_price = grocery[current_shop]["milk"] * costumer.milk()
+        total_bread_price = grocery[current_shop]["bread"] * costumer.bread()
+        total_butter_price = (
+            grocery[current_shop]["butter"] * costumer.butter()
+        )
```

Comment:
Remove this hardcode
If i provide another file with same structure your code will broke